### ISSUE TITLE
Document how Federated queries can work with string matching

### ISSUE THREAD

**Issue by CFGrote** [sparna-git/Sparnatural#605]
If I understand correctly, setting up a federated query requires that the subject node of the `config-core:sparqlService` annotated property exists both in the local graph and in the remote (federated) graph. E.g. in the [documented example](http://docs.sparnatural.eu/Federated-querying.html#and-tadam-), the node `Museum_Wikidata` exists in the local graph and in Wikidata.

In my case, local and remote graphs only share common Literals, not URIs.

For example, my local graph has
```
local:mynode local:crossref_id "abc123" .
```

and the remote graph has
```
remote:somenode remote:id "abc123" .
```

and a query to cross link these two would hence read

```
select * where {
   ?l_node local:crossref_id ?local_id .

   service <http://remote.net/sparql> {
     ?r_node a remote:someClass;
                    remote:id ?remote_id .
    }
    values ?local_id {?remote_id} .
```
or something similar.

My question is if (and how) such a query could be configured in Sparnatural. I hope my example is clear enough.  Thanks for any hint.

**Comment by tfrancart** [comment]
> If I understand correctly, setting up a federated query requires that the subject node of the config-core:sparqlService annotated property exists both in the local graph and in the remote (federated) graph.

I think it could work using inverse property paths. There is a subtlety, let me rephrase : it requires that the node **in the subject position (in terms of query writing)** of the config-core:sparqlService annotated property exists both in the local graph and in the remote (federated) graph.

I think it could work if you use an inverse property path on the sparqueSrevice-annotated property:

```turtle
ex:myPropertyShape
  sh:path [ sh:inversePath remote:id ] ;  
```

the SPARQL query will then be something like:

```sparql
select ?l_node ?r_node where {
   ?l_node local:crossref_id ?theId .

   service <http://remote.net/sparql> {
    ?theId ^remote:id  ?r_node;
    }
```

could you try this and let me know ?

Also let me know if you have a publicly available deployment !

**Comment by tfrancart** [comment]
I confirm the above approach works. Here is a test with a faxe join between DBPedia and Wikidata on the Library of Congress Authority ID, being a Literal. The twist is to introduce the Literal NodeShape, and attach it a property shape with an inverse path linking it to the Wikidata entry:

![Image](https://github.com/user-attachments/assets/ddc4d4f0-073f-419e-99cf-8b28f6a5e864)

Here is the necessary config :

```turtle
this:Museum a sh:NodeShape;
  sh:order "2"^^xsd:integer;
  volipi:iconName "fa-solid fa-university";
  sh:targetClass dbpedia:Museum;
  sh:nodeKind sh:IRI;
  rdfs:label "Museum"@en, "Musée"@fr;
  sh:description "A DBPedia Museum"@en, "Un Musée DBPedia"@fr;
  sh:property this:Museum_country,
  # snip, other properties
  # the property expressing the literal value
   this:Museum_hasLCID .

# The library of congress Authority ID, A Literal
this:Museum_hasLCID sh:path dbpedia:fake_LCID;
  sh:name "LC Authority ID"@en, "LC Authority ID"@fr;
  # points to a Literal NodeShape
  sh:node this:LCAuthorityIdentifier;
  dash:searchWidget core:NonSelectableProperty .

# A Literal NodeShape
this:LCAuthorityIdentifier a sh:NodeShape;
  volipi:iconName "fa-solid fa-a";
  sh:nodeKind sh:Literal;
  rdfs:label "Library of Congress Authority ID"@en, "Library of Congress Authority ID"@fr;
  # A Literal with a property !
  sh:property this:Wikidata_hasLCAuthorityIdentifier ;
.

# The property being an inverse and linking to the Wikidata item
# Note the use of inversePath
this:Wikidata_hasLCAuthorityIdentifier sh:path [ sh:inversePath wdt:P244 ] ;
  sh:name "is LC Authority ID of"@en, "is LC Authority ID of"@fr;
  sh:nodeKind sh:IRI;
  sh:node this:MuseumWikidata;
  # This is using a SPARQL SERVICE keyword
  core:sparqlService <https://www.wikidata.org> .
```

I will document this in the documentation
